src/stratigraphy/util/description_block_splitter.py

```diff
@@ -128,25 +126,16 @@ def separator_condition(self, last_line: TextLine, current_line: TextLine, curre
         for line in self.geometric_lines:
             line_y_coordinate = (line.start.y + line.end.y) / 2
 
-            line_cuts_lefthandside_of_block = line.start.x < current_block.rect.x0 < line.end.x
+            line_cuts_lefthandside_of_text = (line.start.x < last_line.rect.x0 < line.end.x) and (
+                line.start.x < current_line.rect.x0 < line.end.x
+            )
             is_line_long_enough = (
                 np.abs(line.start.x - line.end.x) > self.length_threshold
             )  # for the block splitting, we only care about x-extension
 
             line_ends_block = last_line_y_coordinate < line_y_coordinate < current_line_y_coordinate
 
-            weak_condition = (
-                (line.start.x - 5 < current_block.rect.x0 and line.end.x > current_block.rect.x0)
-                and (np.abs(line.start.x - line.end.x) > self.length_threshold - 2)
-                and (len(current_block.lines) > 1)
-            )  # if block has at least three lines, we weaken the splitting condition.
-            # It is three lines because the statement means that block.lines has at least two elements.
-            # The third line is current_line
-            # The reason for the splitting is, that if we know that a block is long, the probability for
-            # a missed splitting is higher. Therefore, we weaken the condition. The weakened condition
-            # leads to an improved score as per 21.03.2024.
-
-            if line_ends_block and ((is_line_long_enough and line_cuts_lefthandside_of_block) or weak_condition):
```

Comment:
Do you have any intuition why this is not needed any more? Is it because the bounding boxes of the lines start further right compared to the bounding boxes of the blocks?

Reply:
As discussed on Teams: I don't have a clear intuition, but I did not have a clean intuition either for why this "weak condition" was needed in the first place. A plausible explanation is that in some cases, the bounding box of the material descriptions is incorrectly identified. Now we don't use this bounding box anymore, but rather the more local context of the individual text lines just before and after. Maybe it is this change that removes the need for this additional "weak condition".